reana_workflow_engine_snakemake/executor.py

```diff
@@ -154,36 +143,79 @@ def _get_container_image(self, job: Job) -> str:
             log.info(f"No environment specified, falling back to: {container_image}")
         return container_image
 
-    def _handle_job_status(self, job, status):
+    def _handle_job_status(
+        self, job, job_status: JobStatus, workflow_status: RunStatus
+    ) -> None:
         workflow_uuid = os.getenv("workflow_uuid", "default")
         job_id = job.reana_job_id
-        log.info(f"{status} job: {job_id}")
+        log.info(f"{job_status} job: {job_id}")
         message = {
             "progress": build_progress_message(
-                **{status: {"total": 1, "job_ids": [job_id]}}
+                **{job_status.name: {"total": 1, "job_ids": [job_id]}}
             )
         }
-        status_running = 1
-        status_failed = 3
-        status_mapping = {"finished": status_running, "failed": status_failed}
         self.publisher.publish_workflow_status(
-            workflow_uuid, status_mapping[status], message=message
+            workflow_uuid, workflow_status.value, message=message
         )
 
-    def handle_job_success(self, job):
+    def handle_job_success(self, job) -> None:
         """Override job success method to publish job status."""
         # override handle_touch = True, to enable `touch()` in Snakefiles
         super(ClusterExecutor, self).handle_job_success(
-            job, upload_remote=False, handle_log=False, handle_touch=True
+            job, upload_remote=False, handle_log=False, handle_touch=False
         )
 
-        self._handle_job_status(job, "finished")
+        self._handle_job_status(
+            job, job_status=JobStatus.finished, workflow_status=RunStatus.running
+        )
 
-    def handle_job_error(self, job):
+    def handle_job_error(self, job) -> None:
         """Override job error method to publish job status."""
         super().handle_job_error(job)
 
-        self._handle_job_status(job, "failed")
+        self._handle_job_status(
+            job, job_status=JobStatus.failed, workflow_status=RunStatus.failed
+        )
+
+    def _get_job_status_from_controller(self, job_id: str) -> str:
+        response = self.rjc_api_client.check_status(job_id)
+        return response["status"]
```

Comment:
I think if r-j-c returns 500 error, `response["status"]` will fail and trigger workflow error. If `response.get("status")` is used `None` will be returned and workflow will be stuck in `running` status. Let me check when r-j-c might throw 500 errors.

Reply:
And a fun thing I have just discovered, you cannot use `response.get` cause `response` is not a dictionary. It is the object `Job`.

Reply:
I will use `try/catch` with `AttributeError` to check if the `status` field exists. This is a rare event as it can only be caused by a response that doesn't contain 200 or 404 errors.